service/history/handler.go

```diff
@@ -1558,8 +1556,8 @@ func (h *Handler) GetDLQReplicationMessages(ctx context.Context, request *histor
 		}
 	}
 
-	for _, replicationTaskInfos := range taskInfoPerExecution {
-		go handleTaskInfoPerExecution(replicationTaskInfos)
+	for _, replicationTaskInfos := range taskInfoPerShard {
+		go handleTaskInfoPerShard(replicationTaskInfos)
```

Comment:
Probably we should look into capping concurrency for this handling logic.  Number of shards could potentially be large.  Starting a go routine per shard might become a performance issue.  I'm ok to create a separate task to address this.

Reply:
long term strategy is to use the grpc streaming functionality, which will greatly reduce the complexity

Reply:
#1046